managedtenants/bundles/bundle_builder.py

```diff
@@ -64,9 +65,17 @@ def _push_all(self, bundles):
             return
 
         try:
+            # Optimization - only ensure repo once for an operator dir
+            ensure_repo = {}
             for bundle in bundles:
                 self.log.info(f"Pushing {bundle}.")
-                self.docker_api.push(bundle.image)
+                self.docker_api.push(
+                    bundle.image,
+                    ensure_repo=ensure_repo.get(
+                        bundle.bundle_repo_name(), True
+                    ),
+                )
+                ensure_repo[bundle.bundle_repo_name()] = False
```

Comment:
```suggestion
                repo = bundle.bundle_repo_name()
                self.docker_api.push(
                    bundle.image,
                    ensure_repo=repo not in seen,
                )
                seen.add(repo)
```

Using a set `seen` instead of a `dict` might be a little cleaner.

Reply:
it does look cleaner. changed it !